### PR TITLE
[11][ADD] connector_importer

### PR DESCRIPTION
**Pull request by simahawk**
**TODO**
- [x] add README
- [x] more tests
- [x] fixes and cleanup

### CODE REVIEW

File: connector_importer/models/import_type.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        product.product::product.importer.component.name
+
+    Each line contains a couple model::importer.
+    The model is what you want to import, the importer states
+    the name of the connector component to handle the import for that model.
+
+    The importer machinery will run the imports for all the models declared
+    and will retrieve their specific importerts to execute them.
+    """
+    _name = 'import.type'
```
Comment (mpanarin) [inline connector_importer/models/import_type.py:27]: keep the naming`importer.type`?

File: connector_importer/components/importer.py [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+class RecordImporter(Component):
+    """Importer for records.
+
+    This importer is actually the one that does the real import work.
+    It loads each import records and tries to import them
+    and keep tracks of errored, skipped, etc.
+    See `run` method for detailed information on what it does.
+    """
+
+    _name = 'importer.record'
```
Comment (mpanarin) [inline connector_importer/components/importer.py:59]: imo, this is an extremely confusing name with `import.record`. Maybe it's better to follow general naming and call it `importer.importer` or smth

File: connector_importer/models/record.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    the importer machinery will:
+
+    * retrieve the models to import and their importer
+    * process all records and import them
+    * update recordset info
+
+    When the importer will run, it will read all the records,
+    convert them using connector mappers and do the import.
+    """
+    _name = 'import.record'
```
Comment (mpanarin) [inline connector_importer/models/record.py:33]: well, same

File: connector_importer/tests/test_recordset_importer.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        )
+
+    def _fake_lines(self, count, keys=None):
+        return fake_lines(count, keys=keys or [])
+
+
+class TestRecordsetImporter(TestImporterBase):
+
+    @mock.patch('%s.run_import' % RECORD_MODEL)
+    def test_recordset_importer(self, mocked_run_inport):
```
Comment (mpanarin) [inline connector_importer/tests/test_recordset_importer.py:77]: s/inport/import

**Review by mpanarin — APPROVED** [review]
i'd keep naming same across models to prevent confusion
other than a few naming suggestions LGTM

File: connector_importer/components/importer.py (hunk 2 above)
Comment (simahawk) [reply connector_importer/components/importer.py:59]: this is the importer of the record.
The distinction overall, is this:
`import.` identifies odoo records
`importer.` identifies what has to do w/ importers (importers, mappers, etc)

Comment (simahawk) [reply connector_importer/components/importer.py:59]: worth to mention: I spent a bit of time in finding the right way to name those things... I never came up w/ a good solution w/out messing up w/ extra super-long names :cry: 
But I'm open to suggestions :wink:

File: connector_importer/models/record.py [hunk 5]
```diff
@@ -84,12 +92,12 @@ def debug_mode(self):
 
     @api.multi
     @job
-    def import_record(self, component_name, model_name):
+    def import_record(self, component_name, model_name, is_last_importer=True):
```
Comment (mpanarin) [inline connector_importer/models/record.py:95]: I think it's worth leaving the small note on what does parameter `id_last_importer` does. As don't find it self-explanatory

**Review by mpanarin — APPROVED** [review]

**Comment by mpanarin** [comment]
old tests should be updated to check the tuple of three instead of 2

**Comment by simahawk** [comment]
@mpanarin fixed tnx

File: connector_importer/models/record.py (hunk 5 above)
Comment (simahawk) [reply connector_importer/models/record.py:95]: done

**Review by gurneyalex — APPROVED** [review]

**Pull request closed without merging**